src/firefox/lib/glue.js

```diff
@@ -170,6 +180,58 @@ function setUpConnection(freedom, panel, button) {
       });
     }
   });
+
+
+  // Check if user already has a tab open to the uProxy install page.
+  for (var tab of tabs) {
+    if (isInstallPage(tab.url)) {
+      // Attach our content script to the existing tab.
+      tab.attach({contentScriptFile: contentScriptFile});
+
+      emitPromoIfFound(tab.url);
+    }
+  }
+
+  // Attach a handler to check if a tab is opened in the future with a promo.
+  tabs.on('pageshow', function (tab) {
+    emitPromoIfFound(tab.url);
```

Comment:
I think it's a bit odd we're checking for the promo in so many different ways, any chance of deciding on one and deleting the code for the other methods?  Your call on which you think is better.